File: VB-Examples/09_Charts/CreateBoxAndWhiskerChart.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace CreateBoxAndWhiskerChart
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file "BoxAndWhiskerChart.xlsx" from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\BoxAndWhiskerChart.xlsx")

            ' Get the first worksheet from the workbook
            Dim sheet = workbook.Worksheets(0)

            ' Add a chart to the worksheet
            Dim officeChart = sheet.Charts.Add()

            ' Set the chart title to "Yearly Vehicle Sales"
            officeChart.ChartTitle = "Yearly Vehicle Sales"

            ' Set the chart type to Box and Whisker
            officeChart.ChartType = ExcelChartType.BoxAndWhisker

            ' Set the data range for the chart
            officeChart.DataRange = sheet("A1:E17")

            ' Get the first series of the chart
            Dim seriesA = officeChart.Series(0)

            ' Configure data format options for series A
            seriesA.DataFormat.ShowInnerPoints = False
            seriesA.DataFormat.ShowOutlierPoints = True
            seriesA.DataFormat.ShowMeanMarkers = True
            seriesA.DataFormat.ShowMeanLine = False
            seriesA.DataFormat.QuartileCalculationType = ExcelQuartileCalculation.ExclusiveMedian

            ' Get the second series of the chart
            Dim seriesB = officeChart.Series(1)

            ' Configure data format options for series B
            seriesB.DataFormat.ShowInnerPoints = False
            seriesB.DataFormat.ShowOutlierPoints = True
            seriesB.DataFormat.ShowMeanMarkers = True
            seriesB.DataFormat.ShowMeanLine = False
            seriesB.DataFormat.QuartileCalculationType = ExcelQuartileCalculation.InclusiveMedian

            ' Get the third series of the chart
            Dim seriesC = officeChart.Series(2)

            ' Configure data format options for series C
            seriesC.DataFormat.ShowInnerPoints = False
            seriesC.DataFormat.ShowOutlierPoints = True
            seriesC.DataFormat.ShowMeanMarkers = True
            seriesC.DataFormat.ShowMeanLine = False
            seriesC.DataFormat.QuartileCalculationType = ExcelQuartileCalculation.ExclusiveMedian

            ' Save the workbook to a file named "Boxandwhisker_chart.xlsx"
            workbook.SaveToFile("Boxandwhisker_chart.xlsx")
            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer("Boxandwhisker_chart.xlsx")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub


	End Class
End Namespace

File: CS-Examples/09_Charts/CreateBoxAndWhiskerChart.cs
using System;
using System.Windows.Forms;
using Spire.Xls;
using Spire.Xls.Charts;

namespace CreateBoxAndWhiskerChart
{
	public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a new excel document
            Workbook workbook = new Workbook();

            // Load an excel document from the file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\BoxAndWhiskerChart.xlsx");

            // Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            // Add a new chart
            Chart officeChart = sheet.Charts.Add();

            //Set the chart title
            officeChart.ChartTitle = "Yearly Vehicle Sales";

            // Set chart type as Box and Whisker
            officeChart.ChartType = ExcelChartType.BoxAndWhisker;

            // Set data range in the worksheet
            officeChart.DataRange = sheet["A1:E17"];

            // Box and Whisker settings on first series
            ChartSerie seriesA = officeChart.Series[0];
            seriesA.DataFormat.ShowInnerPoints = false;
            seriesA.DataFormat.ShowOutlierPoints = true;
            seriesA.DataFormat.ShowMeanMarkers = true;
            seriesA.DataFormat.ShowMeanLine = false;
            seriesA.DataFormat.QuartileCalculationType = ExcelQuartileCalculation.ExclusiveMedian;

            // Box and Whisker settings on second series   
            ChartSerie seriesB = officeChart.Series[1];
            seriesB.DataFormat.ShowInnerPoints = false;
            seriesB.DataFormat.ShowOutlierPoints = true;
            seriesB.DataFormat.ShowMeanMarkers = true;
            seriesB.DataFormat.ShowMeanLine = false;
            seriesB.DataFormat.QuartileCalculationType = ExcelQuartileCalculation.InclusiveMedian;

            // Box and Whisker settings on third series   
            ChartSerie seriesC = officeChart.Series[2];
            seriesC.DataFormat.ShowInnerPoints = false;
            seriesC.DataFormat.ShowOutlierPoints = true;
            seriesC.DataFormat.ShowMeanMarkers = true;
            seriesC.DataFormat.ShowMeanLine = false;
            seriesC.DataFormat.QuartileCalculationType = ExcelQuartileCalculation.ExclusiveMedian;

            // Save the file
            workbook.SaveToFile("Boxandwhisker_chart.xlsx");

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer("Boxandwhisker_chart.xlsx");
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }


	}
}
